version: '3.8'

services:
  geoapi:
    image: ${DOCKER_REGISTRY-}geoapi
    container_name: geoAPI
    build:
      context: .
      dockerfile: geoAPI/Dockerfile
    ports:
      - "25000:80"
      - "25001:443"
    environment:
      - ConnectionStrings__DefaultConnection=Server=geoAPI.db;Database=test_db;User Id=test_api;password=notapassword
      - Identity__Authority=http://launchpadsept2020.auth
    depends_on:
      - geoAPI.db

  launchpadsept2020.auth:
    image: ${DOCKER_REGISTRY-}launchpadsept2020auth
    container_name: "launchpadsept2020.auth"
    build:
      context: .
      dockerfile: LaunchpadSept2020.Auth/Dockerfile
    ports:
      - "24000:80"
      - "24001:443"
    environment:
      - ConnectionStrings__DefaultConnection=Server=geoAPI.db;Database=test_db;User Id=test_api;password=notapassword
      - Identity__Authority=http://launchpadsept2020.auth
    depends_on:
      - geoAPI.db

  geoAPI.db:
    image: postgis/postgis
    container_name: geoAPI.db
    restart: on-failure
    environment:
      POSTGRES_PASSWORD: notapassword
      POSTGRES_USER: test_api
      POSTGRES_DB: test_db
    ports:
      - 5432:5432
    volumes:
      - db-data:/var/lib/postgresql/data

volumes:
  db-data: